### PR TITLE
Feature/s3 c 2788 obj retention func tests

### PR DESCRIPTION
**Pull request by dora-korpar**
Adds functional tests for PUT and GET Object Retention APIs

### CODE REVIEW

File: tests/functional/aws-node-sdk/test/object/putRetention.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+const bucketName = 'lockenabledputbucket';
+const unlockedBucket = 'locknotenabledputbucket';
+const objectName = 'putobjectretentionobject';
+
+const retentionConfig = {
+    Mode: 'GOVERNANCE',
+    RetainUntilDate: moment().add(1, 'Days').toISOString(),
+};
+
+function _checkError(err, code, statusCode) {
```
Comment (miniscruff) [inline tests/functional/aws-node-sdk/test/object/putRetention.js]: not sure this needs an underscore, also maybe a name like `matchErrorCodes`. It is not entirely clear what if you are checking here.

Comment (miniscruff) [reply tests/functional/aws-node-sdk/test/object/putRetention.js]: Also, it is a copy of the other check error, might want to move it to a util file if it is appropriate.

**Review by miniscruff — APPROVED** [review]

**Comment by miniscruff** [comment]
Generally not a huge fan of after the fact / separate test only changes. Also not entirely sure why these are functionally tests where the other ones were unit tests.

**Review by miniscruff — APPROVED** [review]

**Review by ilkescality — APPROVED** [review]

**Comment by dora-korpar** [comment]
@bert-e reset

**Comment by dora-korpar** [comment]
@bert-e reset

**Comment by dora-korpar** [comment]
@bert-e reset

**Comment by dora-korpar** [comment]
@bert-e approve

**Pull request merged**